src/ann_solo/reader.py

```diff
@@ -186,39 +227,61 @@ def get_spectrum(self, spec_id: int, process_peaks: bool = False)\
         Returns
         -------
         Spectrum
-            The spectrum from the spectral library file with the specified
+            The spectrum from the spectral library store with the specified
             identifier.
         """
-        spectrum = self._parser.read_spectrum(
-            self.spec_info['offset'][spec_id])[0]
+        spectrum = self._spectral_library_store.read_spectrum_from_library(
+                                                spec_id)
         spectrum.is_processed = False
         if process_peaks:
-            process_spectrum(spectrum, True)
-
+            annotation = spectrum.annotation
+            spectrum = process_spectrum(spectrum, True)
+            spectrum._annotation = annotation
```

Comment:
**praise:** Good fix. Apologies for the inflexible spectrum_utils functionality. 🙂